control-plane/subcommand/api-gateway/command.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apigateway
+
+import (
+	"context"
+	"flag"
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/common"
+	"github.com/hashicorp/consul-k8s/control-plane/subcommand/flags"
+	"github.com/mitchellh/cli"
+	"io"
+	"sync"
+)
+
+const (
+	defaultSDSServerHost = "consul-api-gateway-controller.default.svc.cluster.local"
+	defaultSDSServerPort = 9090
+)
+
+type Command struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+
+	flagServerCAFile            string // CA File for CA for Consul server
+	flagServerCASecret          string // CA Secret for Consul server
+	flagServerCASecretNamespace string // CA Secret namespace for Consul server
+
+	flagConsulAddress string // Consul server address
+
+	flagPrimaryDatacenter string // Primary datacenter, may or may not be the datacenter this controller is running in
+
+	flagSDSServerHost string // SDS server host
+	flagSDSServerPort int    // SDS server port
+	flagMetricsPort   int    // Port for prometheus metrics
+	flagPprofPort     int    // Port for pprof profiling
+	flagK8sContext    string // context to use
+	flagK8sNamespace  string // namespace we're run in
+
+	// Consul namespaces
+	flagConsulDestinationNamespace string
+	flagMirrorK8SNamespaces        bool
+	flagMirrorK8SNamespacePrefix   string
+
+	// Logging
+	flagLogLevel string
+	flagLogJSON  bool
+
+	once   sync.Once
+	output io.Writer
+	help   string
+	ctx    context.Context
+}
+
+// New returns a new API Gateway command.
+func New(ctx context.Context, ui cli.Ui, logOutput io.Writer) *Command {
+	return &Command{UI: ui, output: logOutput, ctx: ctx}
+}
+
+func (c *Command) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(&c.flagServerCAFile, "ca-file", "", "Path to CA for Consul server.")
+	c.flags.StringVar(&c.flagServerCASecret, "ca-secret", "", "CA Secret for Consul server.")
+	c.flags.StringVar(&c.flagServerCASecretNamespace, "ca-secret-namespace", "default", "CA Secret namespace for Consul server.")
+	c.flags.StringVar(&c.flagConsulAddress, "consul-address", "", "Consul Address.")
```

Comment:
[Some other commands](https://github.com/hashicorp/consul-k8s/blob/main/control-plane/subcommand/create-federation-secret/command.go#L44) use special types for common flags, however these types have different actual names for the flags. i.e., consul-address just becomes `addresses`. I didn't want to start changing the flags that the user would pass in until we made sure the tradeoff on these changes were worth it, which I think lines up with the guidance from @t-eckert

Reply:
I think while we have the ability to run either binary, it's best to keep these the same. When we get to version 1.3 of Consul K8s, we can start making changes because the old implementation will no longer be used.

Reply:
Okay cool.